dotnet/samples/KernelSyntaxExamples/Example51_StepwisePlanner.cs

```diff
@@ -12,77 +15,213 @@
+
+    internal static string? Suffix = null;
+
     public static async Task RunAsync()
     {
         string[] questions = new string[]
         {
-            "Who is the current president of the United States? What is his current age divided by 2",
-            // "Who is Leo DiCaprio's girlfriend? What is her current age raised to the (his current age)/100 power?",
-            // "What is the capital of France? Who is that city's current mayor? What percentage of their life has been in the 21st century as of today?",
-            // "What is the current day of the calendar year? Using that as an angle in degrees, what is the area of a unit circle with that angle?"
+            "What color is the sky?",
+            "What is the weather in Seattle?",
+            "What is the tallest mountain on Earth? How tall is it divided by 2?",
+            "What is the capital of France? Who is that city's current mayor? What percentage of their life has been in the 21st century as of today?",
+            "What is the current day of the calendar year? Using that as an angle in degrees, what is the area of a unit circle with that angle?",
+            "If a spacecraft travels at 0.99 the speed of light and embarks on a journey to the nearest star system, Alpha Centauri, which is approximately 4.37 light-years away, how much time would pass on Earth during the spacecraft's voyage?"
         };
 
         foreach (var question in questions)
         {
-            var kernel = GetKernel();
-            await RunWithQuestion(kernel, question);
+            for (int i = 0; i < 1; i++)
+            {
+                await RunTextCompletion(question);
+                await RunChatCompletion(question);
+            }
+        }
+
+        PrintResults();
+    }
+
+    // print out summary table of ExecutionResults
+    private static void PrintResults()
+    {
+        Console.WriteLine("**************************");
+        Console.WriteLine("Execution Results Summary:");
+        Console.WriteLine("**************************");
+
+        foreach (var question in ExecutionResults.Select(s => s.question).Distinct())
+        {
+            Console.WriteLine("Question: " + question);
+            Console.WriteLine("Mode\tModel\tAnswer\tStepsTaken\tIterations\tTimeTaken");
+            foreach (var er in ExecutionResults.OrderByDescending(s => s.model).Where(s => s.question == question))
+            {
+                Console.WriteLine($"{er.mode}\t{er.model}\t{er.stepsTaken}\t{er.iterations}\t{er.timeTaken}\t{er.answer}");
+            }
         }
     }
 
-    private static async Task RunWithQuestion(IKernel kernel, string question)
+    private struct ExecutionResult
     {
+        public string mode;
+        public string? model;
+        public string? question;
+        public string? answer;
+        public string? stepsTaken;
+        public string? iterations;
+        public string? timeTaken;
```

Comment:
nit: Should it be PascalCase to follow naming convention?

Reply:
Thanks Dmytro, filing a follow up task for these pieces.